lib/galaxy/config/__init__.py

```diff
@@ -100,11 +100,24 @@
 
 
 def find_root(kwargs):
-    root = os.path.abspath(kwargs.get('root_dir', '.'))
-    return root
+    return os.path.abspath(kwargs.get('root_dir', '.'))
 
 
 class BaseAppConfiguration(object):
+    # Override in subclasses (optional): {KEY: config option, VALUE: deprecated directory name}
+    # If VALUE == first directory in a user-supplied path that resolves to KEY, it will be stripped from that path
+    deprecated_dirs = None
```

Comment:
```suggestion
    deprecated_dirs = {}
```

Reply:
It's only called here: https://github.com/galaxyproject/galaxy/blob/163ba59bb189cbb3bc42c7f7404ee075a6da8aeb/lib/galaxy/config/__init__.py#L210 - and the code checks if the variable has been set. I don't foresee any other use cases for it, so I think `deprecated_dirs = None` is a more straightforward way to say "you ain't gonna need it, unless you know you need it". Whereas an empty dict implies the possibility of loading stuff into it. 
That said, if you think such a safeguard is a reasonable precaution, I'll make it a dict as per Nicola's suggesion.

Reply:
Here as well: https://github.com/galaxyproject/galaxy/pull/9371/files#diff-46f86ea0d82b84642e4218279ef61442R196 but I see that's covered by the code you linked.

Reply:
Yep, that's the one I was referring to.